From c1d7c35b6d2c8b6ec69b90bac6febf673d04acc5 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Mon, 13 Jun 2016 14:36:08 +0800
Subject: [PATCH 1675/1691] MLK-12898: ov5640 mipi: Add more delay to wait
 sensor stable

commit c1d7c35b6d2c8b6ec69b90bac6febf673d04acc5 from
git://git.freescale.com/imx/linux-2.6-imx.git imx_4.1.15_1.0.0_ga

Add more delay to wait sensor stable.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
---
 drivers/media/platform/mxc/subdev/ov5640_mipi.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/media/platform/mxc/subdev/ov5640_mipi.c b/drivers/media/platform/mxc/subdev/ov5640_mipi.c
index ca9156d..739a24c 100644
--- a/drivers/media/platform/mxc/subdev/ov5640_mipi.c
+++ b/drivers/media/platform/mxc/subdev/ov5640_mipi.c
@@ -1106,12 +1106,9 @@ static int ov5640_init_mode(enum ov5640_frame_rate frame_rate,
 		/* dump the first two frames: 1/7.5*2
 		 * the frame rate of QSXGA is 7.5fps */
 		msec_wait4stable = 267;
-	} else if (frame_rate == ov5640_15_fps) {
-		/* dump the first nine frames: 1/15*9 */
+	} else {
+		/* dump the first eighteen frames: 1/30*18 */
 		msec_wait4stable = 600;
-	} else if (frame_rate == ov5640_30_fps) {
-		/* dump the first nine frames: 1/30*9 */
-		msec_wait4stable = 300;
 	}
 	msleep(msec_wait4stable);
 
-- 
1.9.1

